src/quorum/majority.rs

```diff
@@ -147,3 +224,123 @@ impl DerefMut for Configuration {
         &mut self.voters
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::quorum::{AckIndexer, Index};
+    use crate::{HashSet, JointConfig};
+
+    #[test]
+    fn test_describe() {
+        let voters: HashSet<_> = [1, 2, 3].iter().cloned().collect();
+        let c = JointConfig::new(voters);
+        let mut l = AckIndexer::default();
+        l.insert(
+            1,
+            Index {
+                index: 100,
+                group_id: 0,
+            },
+        );
+        l.insert(
+            2,
+            Index {
+                index: 101,
+                group_id: 0,
+            },
+        );
+        l.insert(
+            3,
+            Index {
+                index: 99,
+                group_id: 0,
+            },
+        );
+        assert_eq!(
+            r#"       idx
+x>     100    (id=1)
+xx>    101    (id=2)
+>       99    (id=3)
+"#,
+            c.describe(&l)
+        );
+        l.remove(&1);
+        assert_eq!(
+            r#"       idx
+?        0    (id=1)
+xx>    101    (id=2)
+x>      99    (id=3)
+"#,
+            c.describe(&l)
+        );
+        l.remove(&3);
+        assert_eq!(
+            r#"       idx
+?        0    (id=1)
+xx>    101    (id=2)
+?        0    (id=3)
+"#,
+            c.describe(&l)
+        );
+        l.insert(
+            2,
+            Index {
+                index: 120,
+                group_id: 0,
+            },
+        );
+        assert_eq!(
+            r#"       idx
+?        0    (id=1)
+xx>    120    (id=2)
+?        0    (id=3)
+"#,
+            c.describe(&l)
+        );
+        l.insert(
+            4,
+            Index {
+                index: 120,
+                group_id: 0,
+            },
+        );
+        assert_eq!(
+            r#"       idx
+?        0    (id=1)
+xx>    120    (id=2)
+?        0    (id=3)
+"#,
+            c.describe(&l)
+        );
+        l.insert(
+            3,
+            Index {
+                index: 120,
+                group_id: 1,
+            },
+        );
+        assert_eq!(
+            r#"       idx
+?        0    (id=1)
+x>     120    (id=2)
+>    [1]120    (id=3)
```

Comment:
Because the width in etcd is 5, so this cant be align if the width of id if over 5 (including group id information)